### PR TITLE
fix #2426 Rename Emission to EmitResult

### LINKED ISSUE #2426: Rename `Emission` to `EmitResult`
## Motivation
`Emission` name implies that the class represents a _command_, whereas it is simply an enum that represent the state / result of the `tryEmit` calls.

## Desired solution
Rename to `EmitResult`.

## Open questions
For consistency, `EmissionException` should also probably be renamed to `EmitException`.
But it is less misleading than on the enum, so in the spirit of limiting the impact of this late change, we could keep the exception as is.

### PR DESCRIPTION
**Pull request by simonbasle**

### CODE REVIEW

File: reactor-core/src/main/java/reactor/core/publisher/Sinks.java [hunk 1]
```diff
@@ -86,13 +86,14 @@ public static ManySpec many() {
...
 
 	/**
-	 * Represents the immediate status of a signal emission. This does not guarantee that a signal is consumed,
-	 * it simply refers to the sink state when an emit method is invoked. This is a particularly important
-	 * distinction with regards to {@link #FAIL_CANCELLED} which means the sink is -now- interrupted and emission can't
-	 * proceed. Consequently, it is possible to emit a signal and obtain an "OK" status even if an in-flight cancellation
-	 * is happening. This is due to the async nature of these actions: producer emits while consumer can interrupt independently.
+	 * Represents the immediate result of an emit attempt (eg. in {@link Sinks.Many#tryEmitNext(Object)}.
+	 * This does not guarantee that a signal is consumed, it simply refers to the sink state when an emit method is invoked.
+	 * This is a particularly important distinction with regards to {@link #FAIL_CANCELLED} which means the sink is -now-
```
Comment (ericbottard) [inline reactor-core/src/main/java/reactor/core/publisher/Sinks.java:91]:
```suggestion
	 * This is a particularly important distinction with regard to {@link #FAIL_CANCELLED} which means the sink is -now-
```

File: reactor-core/src/main/java/reactor/core/publisher/Sinks.java [hunk 2]
```diff
@@ -594,56 +595,56 @@ public Emission getReason() {
...
 		 * Try emitting a non-null element, generating an {@link Subscriber#onNext(Object) onNext} signal.
-		 * The result of the attempt is represented as an {@link Emission}, which possibly indicates error cases.
+		 * The result of the attempt is represented as an {@link EmitResult}, which possibly indicates error cases.
 		 * <p>
 		 * Might throw an unchecked exception in case of a fatal error downstream which cannot
 		 * be propagated to any asynchronous handler (aka a bubbling exception).
 		 *
 		 * @param t the value to emit, not null
-		 * @return {@link Emission}
+		 * @return {@link EmitResult}
```
Comment (ericbottard) [inline reactor-core/src/main/java/reactor/core/publisher/Sinks.java:604]: These do not provide any added value, users can already discover the return type.

File: reactor-core/src/main/java/reactor/core/publisher/Sinks.java [hunk 3]
```diff
@@ -721,24 +722,24 @@ public Emission getReason() {
 
 		/**
 		 * Try to complete the {@link Mono} without a value, generating only an {@link Subscriber#onComplete() onComplete} signal.
-		 * The result of the attempt is represented as an {@link Emission}, which possibly indicates error cases.
+		 * The result of the attempt is represented as an {@link EmitResult}, which possibly indicates error cases.
 		 *
-		 * @return {@link Emission}
+		 * @return {@link EmitResult}
```
Comment (ericbottard) [inline reactor-core/src/main/java/reactor/core/publisher/Sinks.java:727]: Same as before, this can be removed if there is nothing to add.

**Review by bsideup — APPROVED** [review]
👍  nothing to add other than @ericbottard's suggestion

**Review by aneveu — APPROVED** [review]

**Comment by simonbasle** [comment]
actually, I'm looking at `hasSucceeded` and `EmitResult#hasSucceeded` reads wrong. `isSuccess()` would be more consistent, wouldn't it ? cc @ericbottard @bsideup

**Comment by bsideup** [comment]
@simonbasle bring it on! 😅 would increase the changeset, but since we're touching the class, I think this change is a good idea

File: reactor-core/src/main/java/reactor/core/publisher/Sinks.java (hunk 2 above)
Comment (simonbasle) [reply reactor-core/src/main/java/reactor/core/publisher/Sinks.java]: improved the comments

**Review by aneveu — APPROVED** [review]

**Pull request merged**